modules/bridge/nd/client.lua

```diff
@@ -1,7 +1,41 @@
-RegisterNetEvent("ND:setCharacter", function(character)
-    local groups = {}
-    for group, info in pairs(character.groups) do
-        groups[group] = info.lvl
+function reorderGroups(groups)
+    groups = groups or {}
+    for group, info in pairs(groups) do
+        groups[group] = info.rank
     end
+    return groups
+end
+
+SetTimeout(500, function()
+	NDCore = exports["ND_Core"]:GetCoreObject()
+		
+	local character = NDCore.Functions.GetSelectedCharacter()
+    if character then
+    	local groups = reorderGroups(character.data.groups)
+    	OnPlayerData("groups", groups)
+	end
+end)
+
+RegisterNetEvent("ND:setCharacter", function(character)
+    local groups = reorderGroups(character.data.groups)
     OnPlayerData("groups", groups)
-end)
\ No newline at end of file
+end)
+
+RegisterNetEvent("ND:jobChanged", function(job, lastJob)
+    local character = NDCore.Functions.GetSelectedCharacter()
+    local groups = reorderGroups(character.data.groups)
+    groups[lastJob.name] = nil
+    groups[job.name] = job.rank
+    
+    OnPlayerData("groups", groups)
+end)
```

Comment:
You have multi-groups but only update "job", and remove lastJob?

Reply:
when job is set/changed then it will also set a group as that job